packages/reactive_ble_mobile/ios/Classes/BleData extras/CharacteristicInstance.swift

```diff
@@ -0,0 +1,99 @@
+import class CoreBluetooth.CBUUID
+import class CoreBluetooth.CBCharacteristic
+import class CoreBluetooth.CBService
+
+struct CharacteristicInstance: Equatable {
+
+    let id: CharacteristicID
+    let instanceID: CharacteristicInstanceID
+    let serviceID: ServiceID
+    let serviceInstanceID: ServiceInstanceID
+    let peripheralID: PeripheralID
+}
+
+extension CharacteristicInstance {
+    
+    init(_ characteristic: CBCharacteristic) throws {
+        guard let service = characteristic.service
+        else {
+            throw Failure.serviceNotFound
+        }
+        
+        guard let peripheral = service.peripheral
+        else {
+            throw Failure.peripheralNotFound
+        }
+        
+        guard
+            let characteristicIndex = service.characteristics?.filter({ c in c.uuid == characteristic.uuid }).index(of: characteristic)
+        else {
+            throw Failure.characteristicNotFound
+        }
+        
+        guard
+            let serviceIndex = peripheral.services?.filter({ s in s.uuid == service.uuid }).index(of: service)
+        else {
+            throw Failure.serviceNotFound
+        }
+        
+        self.init(
+            id: characteristic.uuid,
+            instanceID: "\(characteristicIndex)",
```

Comment:
It looks like here you go from integer to InstanceID, and then back to integer where you need to use it. Is it a good reason for it to ever be a InstanceID? Just asking that because in the place where you convert it back to Int it is not completly obvious why you can do that.

Reply:
I made it a string because I don't know what would be a convenient format on other platforms. Within both the Android and iOS implementation it is known to be (convertible to) an int.